components/LocationList.js

```diff
@@ -1,22 +1,71 @@
 import Card from "./Card";
-import styled from "styled-components";
-import Link from "next/link";
-import { StyledSection } from "./StyledSection";
+import { StyledSection, StyledRowSection } from "./StyledSections";
+import { StyledEditButton } from "./StyledButtons";
+import useLocalStorageState from "use-local-storage-state";
+import { FormContainer, Label, Input } from "./StyledFormElements";
+import { StyledEditModal } from "./StyledEditModal";
+import { StyledLink } from "./StyledLinks";
 
-export const StyledLink = styled(Link)`
-  text-decoration: none;
-`;
+export default function LocationList({ locations, setLocations }) {
+  const [showModal, setShowModal] = useLocalStorageState("showModal", {
+    defaultValue: false,
+  });
+
+  const [editingLocation, setEditingLocation] = useLocalStorageState(
+    "editingLocation",
+    { defaultValue: null }
+  );
+
+  function handleEditClick(location) {
+    setEditingLocation(location);
+    setShowModal(true);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const updatedLocations = [...locations];
+
+    const index = updatedLocations.findIndex(
+      (location) => location.id === editingLocation.id
+    );
+
+    updatedLocations[index].locationName = event.target.locationName.value;
+
+    setLocations(updatedLocations);
```

Comment:
this does work, but it can be solved with just one function called as well:

```suggestion
    setLocations(
      locations.map((location) => {
        return location.id === editingLocation.id
          ? { ...location, locationName: event.target.locationName.value }
          : location;
      })
```